tuf/data/types.go

```diff
@@ -181,6 +182,34 @@ type FileMeta struct {
 	Custom *json.RawMessage `json:"custom,omitempty"`
 }
 
+// Equals returns true if the other FileMeta object is equivalent to this one
+func (f FileMeta) Equals(o FileMeta) bool {
+	if o.Length != f.Length || len(f.Hashes) != len(f.Hashes) {
+		return false
+	}
+	if f.Custom == nil && o.Custom != nil || f.Custom != nil && o.Custom == nil {
```

Comment:
ugh, yeah. I was thinking of the `RawMessage` being a `[]byte` but we essentially have double indirection here. I remember we specifically put that in because of the whole "addressability" thing cause the `RawMessage` to get double base64 encoded, if that's not necessary any more (I feel like I remember something addressing RawMessage and this problem being merged in Go 1.7, possible 1.6) we could also make `Custom` a non-pointer `RawMessage`.